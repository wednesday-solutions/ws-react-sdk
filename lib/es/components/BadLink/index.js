import"antd/lib/button/style";import e from"antd/lib/button";import{_ as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import n from"react";import t from"prop-types";import o from"styled-components";import m from"../T/index.js";import i from"../Warning/index.js";import{c as a}from"../colors-bf7c26a7.js";import{f as l}from"../fonts-56bad7cc.js";import{m as p}from"../media-54d0e07e.js";import"react-intl";import"../If/index.js";var s,d,c,f,g=o.div(s||(s=r(["\n  text-align: center;\n"]))),u=o.img(d||(d=r(["\n  margin-top: 9rem;\n  ","\n  ","\n"])),p.mobile.max("\n    width:75%;\n    margin-top: 12rem;\n  "),p.tablet.max("\n    width:75%;\n    margin-top: 10rem;\n  ")),b=o(m)(c||(c=r(["\n  ",";\n  color: ",";\n  margin: 0rem;\n  ","\n"])),l.style.standard(),a.pawlyBlue,p.largeMobile.max("\n  font-size:0.75rem;\n  ")),x=o(e)(f||(f=r(["\n  border-radius: 6px;\n  margin-top: 2rem;\n  ","\n"])),p.largeMobile.max("\n  text-align: center;\n  "));function j(e){var{noPage:r}=e;return n.createElement("div",null,n.createElement(g,null,n.createElement(u,{src:r}),n.createElement(i,{id:"clawful"}),n.createElement(b,{id:"page_not_found"}),n.createElement(x,{type:"primary"},"Take Me Home")))}j.propTypes={noPage:t.element.isRequired};export default j;export{j as BadLink};
