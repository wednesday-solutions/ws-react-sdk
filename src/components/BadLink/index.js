/**
 *
 * BadLink
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import T from '@components/T';
import Warning from '@components/Warning';
import { colors, fonts, media } from '@themes';
import { Button } from 'antd';

const MainDiv = styled.div`
  text-align: center;
`;

const ClawImage = styled.img`
  margin-top: 9rem;
  ${media.mobile.max(`
    width:75%;
    margin-top: 12rem;
  `)}
  ${media.tablet.max(`
    width:75%;
    margin-top: 10rem;
  `)}
`;

const Slow = styled(T)`
  ${fonts.style.standard()};
  color: ${colors.pawlyBlue};
  margin: 0rem;
  ${media.largeMobile.max(`
  font-size:0.75rem;
  `)}
`;
const TryButton = styled(Button)`
  border-radius: 6px;
  margin-top: 2rem;
  ${media.largeMobile.max(`
  text-align: center;
  `)}
`;
export function BadLink({ noPage }) {
  return (
    <div>
      <MainDiv>
        <ClawImage src={noPage} />
        <Warning id="clawful" />
        <Slow id="page_not_found" />
        <TryButton type="primary">Take Me Home</TryButton>
      </MainDiv>
    </div>
  );
}
BadLink.propTypes = {
  noPage: PropTypes.element.isRequired
};
export default BadLink;
