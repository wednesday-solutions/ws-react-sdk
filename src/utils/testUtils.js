import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
