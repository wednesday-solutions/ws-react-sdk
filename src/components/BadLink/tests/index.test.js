/**
 *
 * Tests for BadLink
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import BadLink from '../index';

describe('<BadLink />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<BadLink />);
    expect(baseElement).toMatchSnapshot();
  });
});
