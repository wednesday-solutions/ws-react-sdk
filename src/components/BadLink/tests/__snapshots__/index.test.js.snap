// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BadLink /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="sc-dlfnbm hWxGDI"
      >
        <img
          class="sc-hKgILt bmiPUu"
        />
        <p
          class="sc-bdfBwQ bZojWc sc-gsTCUz cWOkuF"
          data-testid="warning"
        >
          clawful
        </p>
        <p
          class="sc-bdfBwQ bZojWc sc-eCssSg dfoiiY"
          data-testid="t"
        >
          page_not_found
        </p>
        <button
          class="ant-btn ant-btn-primary sc-jSgupP hnNccu"
          type="button"
        >
          <span>
            Take Me Home
          </span>
        </button>
      </div>
    </div>
  </div>
</body>
`;
